CRM/Core/BAO/Cache.php

```diff
@@ -390,4 +390,35 @@ protected static function whereCache($group, $path, $componentID) {
     return $clauses ? implode(' AND ', $clauses) : '(1)';
   }
 
+  /**
+   * Normalize a cache key.
+   *
+   * This bridges an impedance mismatch between our traditional caching
+   * and PSR-16 -- PSR-16 accepts a narrower range of cache keys.
+   *
+   * @param string $key
+   *   Ex: 'ab/cd:ef'
+   * @return string
+   *   Ex: '_abcd1234abcd1234' or 'ab_xx/cd_xxef'.
+   *   A similar key, but suitable for use with PSR-16-compliant cache providers.
+   */
+  public static function cleanKey($key) {
```

Comment:
Can we add a test that throws a few strings at this function?

Reply:
Good idea. Done.